src/solver/impls/slepc/slepc.cxx

```diff
@@ -162,11 +162,17 @@ SlepcSolver::SlepcSolver(Options* options) {
              .withDefault(0);
 
   tol = options_ref["tol"].doc("SLEPc tolerance").withDefault(1.0e-6);
-  maxIt = options_ref["maxIt"].doc("Maximum iterations").withDefault(PETSC_DEFAULT);
+  maxIt = options_ref["maxIt"].doc("Maximum iterations").withDefault(-1);
+  if (maxIt == -1) {
+    maxIt = PETSC_DEFAULT;
+  }
```

Comment:
PETSc is so incredibly frustrating sometimes. I got grumpy and asked them to revert this change. This is obviously fine as a workaround from our side, but I'm annoyed we have to do this.